README.md

```diff
@@ -155,6 +157,75 @@ if __name__ == '__main__':
     IOLoop.current().run_sync(main)
 ```
 
+## `notebook-http` Mode
+
+The `KernelGatewayApp.api` command line argument can be set to `notebook-http`
+to allow a properly annotated notebook to be served via HTTP endpoints. A
+notebook cell must be annotated with a single line comment at the beginning
+describing the HTTP endpoint. An example python cell is:
+
+```python
+# GET /hello/world
+print("hello world")
+```
+
+This enables an HTTP GET on `/hello/world`. The [api_intro.ipynb](https://github.com/jupyter-incubator/kernel_gateway/blob/master/etc/api_examples/api_intro.ipynb)
+provides a full example notebook for reference.
+
+### Running
+The minimum number of arguments needed to run in HTTP mode are
+`--KernelGatewayApp.api=notebook-http` and
+`--KernelGatewayApp.seed_uri=some/notebook/file.ipynb`.
+
+You can run the kernel gateway in `notebook-http` mode from the Makefile:
+
+```
+make dev ARGS="--KernelGatewayApp.api='notebook-http' \
+--KernelGatewayApp.seed_uri=/srv/kernel_gateway/etc/api_examples/scotch_api.ipynb"
+```
+
+All of the notebooks in `etc/api_examples` are mounted into
+`/srv/kernel_gateway/etc/api_examples/` and can be run in HTTP mode.
+
+### Processing requests
+
+When your cell is invoked there will be a string variable called `REQUEST`
+available in the scope of your cell's code. This string is a JSON object
+representing various properties of the request. You will need to parse this
+string to access the properties. For example, in python:
+
+```python
+# GET /hello/world
+req = json.loads(REQUEST)
+# do something with req
+```
+
+The REQUEST object currently contains the following properties:
+
+* `body` - The JSON parsed value of the body
+* `args` -  An object with keys representing query parameter names and keys
+representing the values. The value will be an array so a query parameter can be
+specified multiple times.
+* `path` -  An object with keys representing path parameter names and keys
+representing the values for the parameters
+
+Path parameters can be specified when registering an endpoint by appending a `:`
+to a value within the path. For example, in python a path with path params
+`firstName` and `lastName` would be defined as:
+
+```python
+# GET /hello/:firstName/:lastName
+```
+
+### Setting The Response
+
+The response for HTTP mode can be set in two ways. The first is to write values
+to standard output. All standard output during your cell's execution will be
+collected and returned verbatim as the response. The second, which is used if no
+output is collected, will be a map of mime types to values. This map is the value
+of `content.data` in the Jupyter [`execute_result`](http://jupyter-client.readthedocs.org/en/latest/messaging.html#id4)
+message.
+
```

Comment:
Be clearer about what is preferred. Reading it I think it's stdout over rich output, but it should maybe be in a list so people get there's an order of precedence.